addon/components/ember-tableish-headers.js

```diff
@@ -1,91 +1,83 @@
-import { A as emberArray } from '@ember/array';
-import { computed } from '@ember/object';
-import Component from '@ember/component';
-import layout from '../templates/components/ember-tableish-headers';
+import Component from '@glimmer/component';
 import { getOwner } from '@ember/application';
+import { action } from '@ember/object';
+import { tracked } from 'tracked-built-ins';
 
-export default Component.extend({
-  layout,
-  classNames: ['ember-tableish-headers'],
+export default class EmberTableishHeaders extends Component {
+  cols = tracked(Set);
 
-  nonce: computed({
-    get() {
-      const config = getOwner(this).resolveRegistration('config:environment');
+  get nonce() {
+    const config = getOwner(this).resolveRegistration('config:environment');
 
-      const configNonce = config['ember-tableish-csp-nonce'];
-      if (configNonce) {
-        return configNonce;
-      }
+    const configNonce = config['ember-tableish-csp-nonce'];
+    if (configNonce) {
+      return configNonce;
+    }
 
-      const metaNonce = document.querySelector(
-        'meta[name="ember-tableish-csp-nonce"]'
-      );
+    const metaNonce = document.querySelector(
+      'meta[name="ember-tableish-csp-nonce"]'
+    );
 
-      if (metaNonce) {
-        return metaNonce.content;
-      }
+    if (metaNonce) {
+      return metaNonce.content;
+    }
 
-      return null;
-    },
+    return null;
+  }
 
-    set(k, v) {
-      return v;
-    }
-  }),
+  get gridColumns() {
+    // see https://css-tricks.com/preventing-a-grid-blowout for why we need to use minmax
+    const strs = [];
 
-  gridColumns: computed('cols.@each.width', {
-    get() {
-      // see https://css-tricks.com/preventing-a-grid-blowout for why we need to use minmax
-      return this.cols.map(c => `minmax(0, ${c.width})`).join(' ');
+    for (const col of this.cols) {
+      strs.push(`minmax(0, ${col.width})`);
     }
-  }),
 
-  msGridColumns: computed('gridColumns', 'table.columnGap', {
-    get() {
-      const { columnGap } = this.table;
+    return strs.join(' ');
+  }
+
+  get msGridColumns() {
+    const { columnGap } = this.args.table;
+
+    if (columnGap) {
+      const strs = [];
 
-      if (columnGap) {
-        return this.cols
-          .map(c => `minmax(0, ${c.width})`)
-          .join(` ${columnGap} `);
+      for (const col of this.cols) {
+        strs.push(`minmax(0, ${col.width})`);
       }
 
-      return this.gridColumns;
+      return strs.join(` ${columnGap} `);
```

Comment:
Landed here, which still uses a local variable `strs = []`, but eliminates the `forEach` in favor of the for..of